pkg/operator/k8sutil/node.go

```diff
@@ -168,9 +168,27 @@ func NodeMeetsPlacementTerms(node v1.Node, placement rookv1.Placement, ignoreWel
 	if !NodeIsTolerable(node, placement.Tolerations, ignoreWellKnownTaints) {
 		return false, nil
 	}
+	if !NodeHasTopologyKeys(node, placement.TopologySpreadConstraints) {
```

Comment:
I don't think we need to consider the topology constraints in this function. This placement should only be used for a legacy way of node selection with the `storage.nodes` element, which doesn't apply to the spreading of OSDs.

Reply:
Got it. Fixed